### PR TITLE
LPS-163844 Error message is displayed on Object list view when using the fields Author, Modified Date or Created Date as Default Sort in a View

### PR DESCRIPTION
**Pull request by manuelecastro**
[LPS-163844](https://issues.liferay.com/browse/LPS-163844)

**Description:**
Error message is displayed on Object list view when using the fields Modified Date or Created Date as Default Sort in a View.

**Proposed solution:** 
Create function to replace the conflicting SortField fieldName with the appropriate terms. The conflicts are caused because the terms being catched by the StringUtil are "createDate" and "modifiedDate" when they should be "dateCreated" and "dateModified" respectively.

### CODE REVIEW

**Comment by manuelecastro** [comment]
ci:test:relevant

**Comment by manuelecastro** [comment]
ci:test:relevant

**Review by p-albuquerque — CHANGES_REQUESTED** [review]

File: modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/sort/SortParserImpl.java [hunk 1]
```diff
@@ -183,6 +183,14 @@ protected boolean isAscending(String orderBy) {
 		return _ASC_DEFAULT;
 	}
 
+	private String _parserFieldSorting(String fieldName) {
```
[the remark this answers is not in the document]
Comment (p-albuquerque) [reply modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/sort/SortParserImpl.java:186]: Also you may consider do it **inline**. Without make a private method.

**Comment by p-albuquerque** [comment]
Feel free to contact me, if you have some ask

**Comment by manuelecastro** [comment]
ci:test:sf

**Comment by manuelecastro** [comment]
ci:test:sf

**Comment by manuelecastro** [comment]
ci:test:relevant

**Comment by manuelecastro** [comment]
ci:test:relevant

**Comment by manuelecastro** [comment]
ci:test:relevant

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/entries/display/context/ViewObjectEntriesDisplayContext.java [hunk 2]
```diff
@@ -199,7 +199,14 @@ public FDSSortItemList getFDSSortItemList() {
 				FDSSortItemBuilder.setDirection(
 					objectViewSortColumn.getSortOrder()
 				).setKey(
-					objectViewSortColumn.getObjectFieldName()
+					() -> {
+						String fieldName = StringUtil.replace(
```
Comment (p-albuquerque) [inline modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/entries/display/context/ViewObjectEntriesDisplayContext.java:203]: About the parameter name. In this kind of situations, we usually link the class name and method name to build a most meaningful variable name

i. e: ObjectField.getName() = objectFieldName

 But in this situation, the method used already have all information of this variable, so you could admit:
 
 objectViewSortColumn.get**ObjectFieldName**() = **objectFieldName**

**Comment by manuelecastro** [comment]
ci:test:sf

**Comment by manuelecastro** [comment]
ci:test:relevant

**Comment by manuelecastro** [comment]
ci:test:relevant

**Comment by manuelecastro** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:forward

**Pull request closed without merging**